core/src/main/java/com/amplifyframework/core/model/AuthStrategy.java

```diff
@@ -25,23 +29,94 @@
      */
-    OWNER,
+    OWNER(Provider.USER_POOLS),
 
     /**
      * Group authorization specifies whether a group can access or operate against an object.  To use GROUPS, the API
      * must have Cognito User Pool configured.
      */
-    GROUPS,
+    GROUPS(Provider.USER_POOLS),
 
     /**
      * The private authorization specifies that everyone will be allowed to access the API with a valid JWT token from
      * the configured Cognito User Pool. To use PRIVATE, the API must have Cognito User Pool configured.
      */
-    PRIVATE,
+    PRIVATE(Provider.USER_POOLS),
 
     /**
      * The public authorization specifies that everyone will be allowed to access the API, behind the scenes the API
      * will be protected with an API Key. To use PUBLIC, the API must have API Key configured.
      */
-    PUBLIC,
+    PUBLIC(Provider.API_KEY);
+
+    private final Provider defaultAuthProvider;
+
+    AuthStrategy(Provider defaultAuthProvider) {
+        this.defaultAuthProvider = defaultAuthProvider;
+    }
+
+    /**
+     * Returns the default provider for the strategy.
+     * @return The default provider for the strategy.
+     */
+    public Provider getDefaultAuthProvider() {
+        return defaultAuthProvider;
+    }
+
+    /**
+     * Represents the the value of the provider field of the @auth directive.
+     */
+    public enum Provider {
+        /**
+         * The userPools provider of the @auth rule directive.
+         */
+        USER_POOLS("userPools"),
+
+        /**
+         * The userPools provider of the @auth rule directive.
+         */
+        OIDC("oidc"),
+
+        /**
+         * The userPools provider of the @auth rule directive.
+         */
+        IAM("iam"),
+
+        /**
+         * The userPools provider of the @auth rule directive.
+         */
+        API_KEY("apiKey");
+
+        private final String authDirectiveProviderName;
+
+        Provider(String authDirectiveProviderName) {
+            this.authDirectiveProviderName = authDirectiveProviderName;
+        }
+
+        /**
+         * Returns the provider name used in the @auth rule directive to represent the current enum item.
+         * @return The name associated with the enum item.
+         */
+        public String getAuthDirectiveProviderName() {
+            return this.authDirectiveProviderName;
+        }
+
+        /**
+         * Retrieve the enum value for the given name.
+         * @param authDirectiveProviderName String value of the provider attribute of the @auth directive.
+         * @return The associated {@link Provider} item associated with the name.
+         * @throws IllegalArgumentException If the value provided in the parameter does not
+         * have a matching element in the enum.
+         */
+        public static Provider from(@NonNull String authDirectiveProviderName) {
+            Objects.requireNonNull(authDirectiveProviderName);
+            for (Provider provider : Provider.values()) {
```

Comment:
Where does `Provider` come from?

Reply:
`Provider` in this case is the enum that this function belongs to. `Provider.values()` returns a list of the enum items to iterate through. This function is essentially converts a string value in the appropriate enum item to represent the auth rule provider, by iterating through those values.

Reply:
Got it!